### PR TITLE
Use subunit-trace

### PR DESCRIPTION
**Pull request by mtreinish**
This adds subunit-trace output filtering to stestr. The code is copied directly from the os-testr project [1]. This enables subunit-trace by default on the run and last commands, and enables it as an option on the load command. Right now most of the subunit-trace output options are hard coded, with the only exception being colorized output which there is a config flag for. In the future we might add support for more customized output.

One thing to note is that to avoid a potential namespace collision this does not add the console_script entrypoint for the subunit-trace command. If/when os-testr switches to using stestr we might be able to figure out a deprecation path in os-testr and remove the forked copy of subunit-trace.

Closes Issue #22 

[1] http://git.openstack.org/cgit/openstack/os-testr

### CODE REVIEW

File: stestr/commands/load.py [hunk 1]
```diff
@@ -116,6 +129,16 @@ def make_tests():
...
+    if pretty_out:
+        outcomes = testtools.StreamToDict(
+            functools.partial(subunit_trace.show_outcome, sys.stdout,
+                              enable_color=color))
+        summary_result = testtools.StreamSummary()
+        output_result = testtools.CopyStreamResult([outcomes, summary_result])
+        output_result = testtools.StreamResultRouter(output_result)
+        cat = subunit.test_results.CatFiles(sys.stdout)
+        output_result.add_rule(cat, 'test_id', test_id=None)
+        start_time = datetime.datetime.utcnow()
```
Comment (masayukig) [inline stestr/commands/load.py]: This causes an error[1] when pretty_out is True(--no-subunit-trace specified).
[1] ```UnboundLocalError: local variable 'start_time' referenced before assignment```

Comment (mtreinish) [inline stestr/commands/load.py]: Yep, start time isn't defined in the case pretty_out is false. So when L157 is run there is no start time defined. I should have put start time definition out one indent level and to L144. I'll respin it right now

**Comment by mtreinish** [comment]
ok, the latest revision should have fixed the issue you caught. Thanks for testing and reviewing this and all the other patches this week.

**Comment by masayukig** [comment]
(I really don't know the github workflow... :-p)

**Pull request merged**